lib/ws_servers/api/algos/algo_worker.js

```diff
@@ -24,25 +26,28 @@ class AlgoWorker {
     this.marketData = marketData
   }
 
-  async start (apiKey, apiSecret, userID) {
-    const { dms, affiliateCode, wsURL, restURL } = this.settings
+  async start ({ apiKey, apiSecret, authToken, userId }) {
+    const { dms, affiliateCode, wsURL, restURL, plugins = [] } = this.settings
 
-    this.userID = userID
+    this.userId = userId
 
     d(
       'spawning bfx algo host (dms %s) [aff %s]',
       dms ? 'enabled' : 'disabled',
       affiliateCode
     )
 
+    if (!authToken) {
+      authToken = await this.generateAuthToken(restURL, apiKey, apiSecret)
+    }
```

Comment:
hmmm... would time out after a few hours... or at maximum after 1 week

something like `setTimeout(refreshToken, ttl - 1hr)` to refresh the tokens on the fly? maybe a modified version of the refresh plugin which connects directly to the api?

Reply:
You're right; for the hosted version, that wouldn't be a problem, but for the UI, it is. But, maintaining in both places is redundant. I'll modify the plugin and delegate this responsibility to the server.

Reply:
fixed: df728cfa6cf076988c6d52a8ad60ed1e11fd951b